src/Init/Coe.lean

```diff
@@ -103,25 +103,66 @@ class Coe (α : Sort u) (β : Sort v) where
   /-- Coerces a value of type `α` to type `β`. Accessible by the notation `↑x`,
```

Comment:
Explain chaining, when to use Coe vs. CoeOut, variable conditions above in module docstring!